### ISSUE TITLE
Move creator metadata population to dandischema

### ISSUE THREAD

**Issue by dchiquito** [dandi/dandi-api#352]
As of #351 we are injecting some creator metadata into the dandiset metadata when it is created. Because this metadata may change based on schema version, we should move that function into dandischema.

**Comment by satra** [comment]
i was hoping that this would happen in "meditor style" on the UI when registering a new dandiset since the UI already puts the out a form based on a version of the schema. hence the form could be a subset of the dandiset schema. this way the api server doesn't need to know anything about this.

**Comment by waxlamp** [comment]
Can you clarify what you mean by "meditor style"? Adding the creator of the dandiset as a contributor/contact person doesn't seem like it needs a human-in-the-loop at all.

**Comment by satra** [comment]
meditor is a form filling interface. new dandiset registration is also a form. we can use the meditor interface to simply ask people to fill a stripped subset of the dandiset schema (name, description, contributors, license as an example).

**Comment by waxlamp** [comment]
Thanks for the clarification.

**Comment by yarikoptic** [comment]
in the light of https://github.com/dandi/dandischema/pull/46 + https://github.com/dandi/dandi-api/pull/364 tandem, establishing a collection of "adapters" within `dandischema` which would simplify and centralize the logic of taking "minimally sufficient" parametrization and returning proper "schema" constructs sounds like a good idea to me ATM.

**Comment by satra** [comment]
indeed, we should be moving things that the api or cli would do to adapters in schema. my point above for this specific thing is that this should happen in the Web client rather than the API. 

we may also need to start building a js version as well unless we can run dandischema in a browser via pyiodide (this may be possible, looking into it).

**Comment by waxlamp** [comment]
I think maybe this is solved? When someone creates a dandiset, their full name is added as the contact person. If I understand the discussion here, there's not a need to move any functionality to dandischema. Can you confirm/clarify, @satra?

**Comment by satra** [comment]
we should consider whatever utilities that work on metadata to be useful to the backend be implemented in dandischema. sometimes it's trivial that it doesn't need to be. but perhaps some functions are indeed best done in the schema. since some of these functions have already been implemented in the api, perhaps easiest is to send a PR to dandischema with those functions ? i also see that #426 refers to this as well.

**Comment by waxlamp** [comment]
Ok, thanks Satra. If you come across other such functions, please file separate issues for them.

**Comment by waxlamp** [comment]
Ah gotcha.

@dchiquito, I've marked this issue as "accepted" and assigned you. Let me know if you'd like to delegate it to someone else instead.
